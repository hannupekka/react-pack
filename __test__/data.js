// Sample repos.
export const repos = {
  12482715: {
    hooks_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/hooks',
    default_branch: 'master',
    branches_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/branches{/branch}',
    size: 144,
    forks_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/forks',
    forks_count: 0,
    fork: false,
    git_url: 'git://github.com/hannupekka/fressi-gym-activity.git',
    private: false,
    keys_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/keys{/key_id}',
    comments_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/comments{/number}',
    trees_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/git/trees{/sha}',
    has_downloads: true,
    blobs_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/git/blobs{/sha}',
    teams_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/teams',
    watchers_count: 0,
    stargazers_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/stargazers',
    stargazers_count: 0,
    contributors_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/contributors',
    tags_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/tags',
    issue_comment_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/comments{/number}',
    events_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/events',
    created_at: '2013-08-30T09:17:39Z',
    issues_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/issues{/number}',
    contents_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/contents/{+path}',
    merges_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/merges',
    name: 'fressi-gym-activity',
    languages_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/languages',
    commits_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/commits{/sha}',
    subscription_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/subscription',
    clone_url: 'https://github.com/hannupekka/fressi-gym-activity.git',
    homepage: null,
    issue_events_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/events{/number}',
    mirror_url: null,
    labels_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/labels{/name}',
    url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity',
    open_issues: 0,
    statuses_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/statuses/{sha}',
    forks: 0,
    archive_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/{archive_format}{/ref}',
    milestones_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/milestones{/number}',
    owner: 4472477,
    assignees_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/assignees{/user}',
    has_wiki: true,
    compare_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/compare/{base}...{head}',
    git_tags_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/git/tags{/sha}',
    updated_at: '2013-10-31T15:13:19Z',
    watchers: 0,
    notifications_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/notifications{?since,all,participating}',
    ssh_url: 'git@github.com:hannupekka/fressi-gym-activity.git',
    language: 'Python',
    pushed_at: '2013-10-31T15:13:15Z',
    downloads_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/downloads',
    subscribers_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/subscribers',
    id: 12482715,
    svn_url: 'https://github.com/hannupekka/fressi-gym-activity',
    full_name: 'hannupekka/fressi-gym-activity',
    html_url: 'https://github.com/hannupekka/fressi-gym-activity',
    description: 'Fetches, parses and formats gym activity for Fressi gyms',
    releases_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/releases{/id}',
    git_refs_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/git/refs{/sha}',
    collaborators_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/collaborators{/collaborator}',
    pulls_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/pulls{/number}',
    deployments_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/deployments',
    has_projects: true,
    has_pages: false,
    open_issues_count: 0,
    git_commits_url: 'https://api.github.com/repos/hannupekka/fressi-gym-activity/git/commits{/sha}',
    has_issues: true
  },
  18128997: {
    hooks_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/hooks',
    default_branch: 'master',
    branches_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/branches{/branch}',
    size: 13335,
    forks_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/forks',
    forks_count: 0,
    fork: true,
    git_url: 'git://github.com/hannupekka/heroku-buildpack-python-libffi.git',
    private: false,
    keys_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/keys{/key_id}',
    comments_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/comments{/number}',
    trees_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/trees{/sha}',
    has_downloads: true,
    blobs_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/blobs{/sha}',
    teams_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/teams',
    watchers_count: 0,
    stargazers_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/stargazers',
    stargazers_count: 0,
    contributors_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/contributors',
    tags_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/tags',
    issue_comment_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/issues/comments{/number}',
    events_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/events',
    created_at: '2014-03-26T06:35:13Z',
    issues_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/issues{/number}',
    contents_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/contents/{+path}',
    merges_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/merges',
    name: 'heroku-buildpack-python-libffi',
    languages_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/languages',
    commits_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/commits{/sha}',
    subscription_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/subscription',
    clone_url: 'https://github.com/hannupekka/heroku-buildpack-python-libffi.git',
    homepage: '',
    issue_events_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/issues/events{/number}',
    mirror_url: null,
    labels_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/labels{/name}',
    url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi',
    open_issues: 0,
    statuses_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/statuses/{sha}',
    forks: 0,
    archive_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/{archive_format}{/ref}',
    milestones_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/milestones{/number}',
    owner: 4472477,
    assignees_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/assignees{/user}',
    has_wiki: true,
    compare_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/compare/{base}...{head}',
    git_tags_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/tags{/sha}',
    updated_at: '2014-03-26T07:28:26Z',
    watchers: 0,
    notifications_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/notifications{?since,all,participating}',
    ssh_url: 'git@github.com:hannupekka/heroku-buildpack-python-libffi.git',
    language: 'Python',
    pushed_at: '2014-03-26T07:28:26Z',
    downloads_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/downloads',
    subscribers_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/subscribers',
    id: 18128997,
    svn_url: 'https://github.com/hannupekka/heroku-buildpack-python-libffi',
    full_name: 'hannupekka/heroku-buildpack-python-libffi',
    html_url: 'https://github.com/hannupekka/heroku-buildpack-python-libffi',
    description: 'A fork of heroku-buildpack-python that includes some additional build dependencies: libffi.  This buildpack is not currently supported or maintained.',
    releases_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/releases{/id}',
    git_refs_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/refs{/sha}',
    collaborators_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/collaborators{/collaborator}',
    pulls_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/pulls{/number}',
    deployments_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/deployments',
    has_projects: true,
    has_pages: false,
    open_issues_count: 0,
    git_commits_url: 'https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/commits{/sha}',
    has_issues: false
  },
};

export const users = {
  4472477: {
    gists_url: 'https://api.github.com/users/hannupekka/gists{/gist_id}',
    following_url: 'https://api.github.com/users/hannupekka/following{/other_user}',
    followers_url: 'https://api.github.com/users/hannupekka/followers',
    subscriptions_url: 'https://api.github.com/users/hannupekka/subscriptions',
    received_events_url: 'https://api.github.com/users/hannupekka/received_events',
    events_url: 'https://api.github.com/users/hannupekka/events{/privacy}',
    avatar_url: 'https://avatars3.githubusercontent.com/u/4472477?v=3',
    login: 'hannupekka',
    url: 'https://api.github.com/users/hannupekka',
    starred_url: 'https://api.github.com/users/hannupekka/starred{/owner}{/repo}',
    organizations_url: 'https://api.github.com/users/hannupekka/orgs',
    repos_url: 'https://api.github.com/users/hannupekka/repos',
    gravatar_id: '',
    site_admin: false,
    type: 'User',
    id: 4472477,
    html_url: 'https://github.com/hannupekka'
  }
};
