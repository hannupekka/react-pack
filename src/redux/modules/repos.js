// @flow
import { fromJS, Map } from 'immutable';
import { Observable, Action } from 'rxjs';
import { ajax } from 'rxjs/observable/dom/ajax';
import { API_HOST } from 'constants/config';

export const FETCH_REPOS = 'react-pack/repos/FETCH_REPOS';
export const FETCH_REPOS_SUCCESS = 'react-pack/repos/FETCH_REPOS_SUCCESS';
export const FETCH_REPOS_FAILURE = 'react-pack/repos/FETCH_REPOS_FAILURE';

export const fetchRepos = (username: string): ThunkAction => ({
  type: FETCH_REPOS,
  payload: { username }
});

export const fetchReposSuccess = (payload: Object): ThunkAction => ({
  type: FETCH_REPOS_SUCCESS,
  payload
});

export const fetchReposEpic = (action$: Observable<Action>): Observable<Action> =>
  action$.ofType(FETCH_REPOS)
    .flatMap(action =>
      ajax.getJSON(`${API_HOST}/users/${action.payload.username}/repos`)
        .flatMap(response => Observable.of(fetchReposSuccess(response)))
        .catch(() => Observable.of({
          type: FETCH_REPOS_FAILURE
        }))
    );

type State = Map<string, any>;
export const initialState: State = fromJS({
  isLoading: false,
  isError: false,
  repos: []
});

export default function reducer(state: State = initialState, action: ThunkAction): State {
  switch (action.type) {
    case FETCH_REPOS:
      return state.merge({ isLoading: true, isError: false, repos: [] });
    case FETCH_REPOS_SUCCESS:
      return state.merge({
        isLoading: false,
        isError: false,
        repos: action.payload
      });
    case FETCH_REPOS_FAILURE:
      return state.merge({ isLoading: false, isError: true });
    default:
      return state;
  }
}
