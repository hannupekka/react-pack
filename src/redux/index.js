// @flow
import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import repos, { fetchReposLogic } from 'redux/repos';
import ui from 'redux/ui';
import users, { fetchRandomUserLogic, fetchSecondRandomUserLogic } from 'redux/users';

export const rootReducer = combineReducers({
  repos,
  ui,
  users,
  router: routerReducer,
});

export const logics = [
  fetchReposLogic,
  fetchRandomUserLogic,
  fetchSecondRandomUserLogic,
];
