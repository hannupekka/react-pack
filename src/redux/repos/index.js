// @flow
import { Observable, Action } from 'rxjs';
import { ajax } from 'rxjs/observable/dom/ajax';
import { normalize } from 'normalizr';
import { repos } from 'redux/repos/schemas';
import { API_HOST } from 'constants/config';

export const FETCH_REPOS = 'react-pack/repos/FETCH_REPOS';
export const FETCH_REPOS_SUCCESS = 'react-pack/repos/FETCH_REPOS_SUCCESS';
export const FETCH_REPOS_FAILURE = 'react-pack/repos/FETCH_REPOS_FAILURE';
export const TOGGLE_SHOW_FORKS = 'react-pack/repos/TOGGLE_SHOW_FORKS';

export const fetchRepos = (username: string): ThunkAction => ({
  type: FETCH_REPOS,
  payload: { username },
});

export const fetchReposSuccess = (payload: Object): ThunkAction => ({
  type: FETCH_REPOS_SUCCESS,
  payload,
});

export const toggleShowForks = (): ThunkAction => ({
  type: TOGGLE_SHOW_FORKS,
  payload: {},
});

export const fetchReposEpic = (action$: Observable<Action>): Observable<Action> =>
  action$.ofType(FETCH_REPOS)
    .flatMap(action =>
      ajax.getJSON(`${API_HOST}/users/${action.payload.username}/repos`)
        .flatMap(response =>
          Observable.of(fetchReposSuccess(normalize(response, repos)))
        )
        .catch(e => Observable.of({
          type: FETCH_REPOS_FAILURE,
          payload: e,
        }))
    );

export const initialState: ReposState = {
  isLoading: false,
  isError: false,
  showForks: true,
  entities: {
    repos: {},
    users: {},
  },
  result: [],
};

export default function reducer(state: ReposState = initialState, action: ThunkAction): ReposState {
  switch (action.type) {
    case FETCH_REPOS:
      return {
        ...state,
        isLoading: true,
      };
    case FETCH_REPOS_SUCCESS:
      return {
        ...state,
        entities: action.payload.entities,
        result: action.payload.result,
        isLoading: false,
      };
    case FETCH_REPOS_FAILURE:
      return {
        ...state,
        isLoading: false,
        isError: true,
        entities: initialState.entities,
        result: initialState.result,
      };
    case TOGGLE_SHOW_FORKS:
      return {
        ...state,
        showForks: !state.showForks,
      };
    default:
      return state;
  }
}
