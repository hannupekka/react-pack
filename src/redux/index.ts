import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import repos, { fetchReposLogic } from '@app/redux/repos';
import ui from '@app/redux/ui';
import users, { fetchRandomUserLogic, fetchSecondRandomUserLogic } from '@app/redux/users';

export const rootReducer = combineReducers({
  repos,
  ui,
  users,
  router: routerReducer,
});

export const logics = [fetchReposLogic, fetchRandomUserLogic, fetchSecondRandomUserLogic];
