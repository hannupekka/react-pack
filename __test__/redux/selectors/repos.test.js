import R from 'ramda';
import configureStore from 'store/configureStore';
import getVisibleRepos from 'redux/repos/selectors';
import { fetchReposSuccess, toggleShowForks } from 'redux/repos';
import { repos, users } from '../../data';

describe('Selector', () => {
  it('should return all repos', () => {
    // Get store.
    const store = configureStore();

    // Mock our repository fetch success data.
    const payload = {
      entities: {
        repos,
        users
      },
      result: R.mapObjIndexed(repo => repo.id, repos)
    };

    // Dispatch fetch success.
    store.dispatch(fetchReposSuccess(payload));

    // Get expected state.
    const expected = getVisibleRepos(store.getState());
    expect(expected).toMatchSnapshot();
  });

  it('should not return forked repos', () => {
    // Get store.
    const store = configureStore();

    // Mock our repository fetch success data.
    const payload = {
      entities: {
        repos,
        users
      },
      result: R.mapObjIndexed(repo => repo.id, repos)
    };

    // Dispatch fetch success and then hide forks.
    store.dispatch(fetchReposSuccess(payload));
    store.dispatch(toggleShowForks());

    // Get expected state.
    const expected = getVisibleRepos(store.getState());
    expect(expected).toMatchSnapshot();
  });
});
