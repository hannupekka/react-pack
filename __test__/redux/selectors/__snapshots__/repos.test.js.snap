// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector should not return forked repos 1`] = `
Object {
  "12482715": Object {
    "archive_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/{archive_format}{/ref}",
    "assignees_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/assignees{/user}",
    "blobs_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/blobs{/sha}",
    "branches_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/branches{/branch}",
    "clone_url": "https://github.com/hannupekka/fressi-gym-activity.git",
    "collaborators_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/collaborators{/collaborator}",
    "comments_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/comments{/number}",
    "commits_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/commits{/sha}",
    "compare_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/compare/{base}...{head}",
    "contents_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/contents/{+path}",
    "contributors_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/contributors",
    "created_at": "2013-08-30T09:17:39Z",
    "default_branch": "master",
    "deployments_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/deployments",
    "description": "Fetches, parses and formats gym activity for Fressi gyms",
    "downloads_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/downloads",
    "events_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/events",
    "fork": false,
    "forks": 0,
    "forks_count": 0,
    "forks_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/forks",
    "full_name": "hannupekka/fressi-gym-activity",
    "git_commits_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/commits{/sha}",
    "git_refs_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/refs{/sha}",
    "git_tags_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/tags{/sha}",
    "git_url": "git://github.com/hannupekka/fressi-gym-activity.git",
    "has_downloads": true,
    "has_issues": true,
    "has_pages": false,
    "has_projects": true,
    "has_wiki": true,
    "homepage": null,
    "hooks_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/hooks",
    "html_url": "https://github.com/hannupekka/fressi-gym-activity",
    "id": 12482715,
    "issue_comment_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/comments{/number}",
    "issue_events_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/events{/number}",
    "issues_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues{/number}",
    "keys_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/keys{/key_id}",
    "labels_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/labels{/name}",
    "language": "Python",
    "languages_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/languages",
    "merges_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/merges",
    "milestones_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/milestones{/number}",
    "mirror_url": null,
    "name": "fressi-gym-activity",
    "notifications_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/notifications{?since,all,participating}",
    "open_issues": 0,
    "open_issues_count": 0,
    "owner": 4472477,
    "private": false,
    "pulls_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/pulls{/number}",
    "pushed_at": "2013-10-31T15:13:15Z",
    "releases_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/releases{/id}",
    "size": 144,
    "ssh_url": "git@github.com:hannupekka/fressi-gym-activity.git",
    "stargazers_count": 0,
    "stargazers_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/stargazers",
    "statuses_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/statuses/{sha}",
    "subscribers_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/subscribers",
    "subscription_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/subscription",
    "svn_url": "https://github.com/hannupekka/fressi-gym-activity",
    "tags_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/tags",
    "teams_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/teams",
    "trees_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/trees{/sha}",
    "updated_at": "2013-10-31T15:13:19Z",
    "url": "https://api.github.com/repos/hannupekka/fressi-gym-activity",
    "watchers": 0,
    "watchers_count": 0,
  },
}
`;

exports[`Selector should return all repos 1`] = `
Object {
  "12482715": Object {
    "archive_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/{archive_format}{/ref}",
    "assignees_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/assignees{/user}",
    "blobs_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/blobs{/sha}",
    "branches_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/branches{/branch}",
    "clone_url": "https://github.com/hannupekka/fressi-gym-activity.git",
    "collaborators_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/collaborators{/collaborator}",
    "comments_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/comments{/number}",
    "commits_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/commits{/sha}",
    "compare_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/compare/{base}...{head}",
    "contents_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/contents/{+path}",
    "contributors_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/contributors",
    "created_at": "2013-08-30T09:17:39Z",
    "default_branch": "master",
    "deployments_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/deployments",
    "description": "Fetches, parses and formats gym activity for Fressi gyms",
    "downloads_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/downloads",
    "events_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/events",
    "fork": false,
    "forks": 0,
    "forks_count": 0,
    "forks_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/forks",
    "full_name": "hannupekka/fressi-gym-activity",
    "git_commits_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/commits{/sha}",
    "git_refs_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/refs{/sha}",
    "git_tags_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/tags{/sha}",
    "git_url": "git://github.com/hannupekka/fressi-gym-activity.git",
    "has_downloads": true,
    "has_issues": true,
    "has_pages": false,
    "has_projects": true,
    "has_wiki": true,
    "homepage": null,
    "hooks_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/hooks",
    "html_url": "https://github.com/hannupekka/fressi-gym-activity",
    "id": 12482715,
    "issue_comment_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/comments{/number}",
    "issue_events_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues/events{/number}",
    "issues_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/issues{/number}",
    "keys_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/keys{/key_id}",
    "labels_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/labels{/name}",
    "language": "Python",
    "languages_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/languages",
    "merges_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/merges",
    "milestones_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/milestones{/number}",
    "mirror_url": null,
    "name": "fressi-gym-activity",
    "notifications_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/notifications{?since,all,participating}",
    "open_issues": 0,
    "open_issues_count": 0,
    "owner": 4472477,
    "private": false,
    "pulls_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/pulls{/number}",
    "pushed_at": "2013-10-31T15:13:15Z",
    "releases_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/releases{/id}",
    "size": 144,
    "ssh_url": "git@github.com:hannupekka/fressi-gym-activity.git",
    "stargazers_count": 0,
    "stargazers_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/stargazers",
    "statuses_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/statuses/{sha}",
    "subscribers_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/subscribers",
    "subscription_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/subscription",
    "svn_url": "https://github.com/hannupekka/fressi-gym-activity",
    "tags_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/tags",
    "teams_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/teams",
    "trees_url": "https://api.github.com/repos/hannupekka/fressi-gym-activity/git/trees{/sha}",
    "updated_at": "2013-10-31T15:13:19Z",
    "url": "https://api.github.com/repos/hannupekka/fressi-gym-activity",
    "watchers": 0,
    "watchers_count": 0,
  },
  "18128997": Object {
    "archive_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/{archive_format}{/ref}",
    "assignees_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/assignees{/user}",
    "blobs_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/blobs{/sha}",
    "branches_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/branches{/branch}",
    "clone_url": "https://github.com/hannupekka/heroku-buildpack-python-libffi.git",
    "collaborators_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/collaborators{/collaborator}",
    "comments_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/comments{/number}",
    "commits_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/commits{/sha}",
    "compare_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/compare/{base}...{head}",
    "contents_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/contents/{+path}",
    "contributors_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/contributors",
    "created_at": "2014-03-26T06:35:13Z",
    "default_branch": "master",
    "deployments_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/deployments",
    "description": "A fork of heroku-buildpack-python that includes some additional build dependencies: libffi.  This buildpack is not currently supported or maintained.",
    "downloads_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/downloads",
    "events_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/events",
    "fork": true,
    "forks": 0,
    "forks_count": 0,
    "forks_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/forks",
    "full_name": "hannupekka/heroku-buildpack-python-libffi",
    "git_commits_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/commits{/sha}",
    "git_refs_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/refs{/sha}",
    "git_tags_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/tags{/sha}",
    "git_url": "git://github.com/hannupekka/heroku-buildpack-python-libffi.git",
    "has_downloads": true,
    "has_issues": false,
    "has_pages": false,
    "has_projects": true,
    "has_wiki": true,
    "homepage": "",
    "hooks_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/hooks",
    "html_url": "https://github.com/hannupekka/heroku-buildpack-python-libffi",
    "id": 18128997,
    "issue_comment_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/issues/comments{/number}",
    "issue_events_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/issues/events{/number}",
    "issues_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/issues{/number}",
    "keys_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/keys{/key_id}",
    "labels_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/labels{/name}",
    "language": "Python",
    "languages_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/languages",
    "merges_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/merges",
    "milestones_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/milestones{/number}",
    "mirror_url": null,
    "name": "heroku-buildpack-python-libffi",
    "notifications_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/notifications{?since,all,participating}",
    "open_issues": 0,
    "open_issues_count": 0,
    "owner": 4472477,
    "private": false,
    "pulls_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/pulls{/number}",
    "pushed_at": "2014-03-26T07:28:26Z",
    "releases_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/releases{/id}",
    "size": 13335,
    "ssh_url": "git@github.com:hannupekka/heroku-buildpack-python-libffi.git",
    "stargazers_count": 0,
    "stargazers_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/stargazers",
    "statuses_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/statuses/{sha}",
    "subscribers_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/subscribers",
    "subscription_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/subscription",
    "svn_url": "https://github.com/hannupekka/heroku-buildpack-python-libffi",
    "tags_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/tags",
    "teams_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/teams",
    "trees_url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi/git/trees{/sha}",
    "updated_at": "2014-03-26T07:28:26Z",
    "url": "https://api.github.com/repos/hannupekka/heroku-buildpack-python-libffi",
    "watchers": 0,
    "watchers_count": 0,
  },
}
`;
